src/sentry/tasks/performance_detection.py

```diff
@@ -0,0 +1,303 @@
+    }
+
+    for span in spans:
+        for _, detector in detectors.items():
+            detector.visit_span(span)
+
+    all_fingerprints = [i for _, d in detectors.items() for i in d.stored_issues]
+
+    if all_fingerprints:
+        sdk_span.set_measurement("_performance_issue_count", len(all_fingerprints))
+        if event_id:
+            sdk_span.set_tag("_performance_issue_transaction_id", event_id)
+
+    duplicate_performance_issues = detectors[DetectorType.DUPLICATE_SPANS].stored_issues
+    duplicate_performance_fingerprints = list(duplicate_performance_issues.keys())
+    if duplicate_performance_fingerprints:
+        first_duplicate = duplicate_performance_issues[duplicate_performance_fingerprints[0]]
+        sdk_span.set_tag("_performance_issue_duplicate_spans", first_duplicate["span_id"])
+
+    slow_span_performance_issues = detectors[DetectorType.SLOW_SPAN].stored_issues
+    slow_performance_fingerprints = list(slow_span_performance_issues.keys())
+    if slow_performance_fingerprints:
+        first_slow_span = slow_span_performance_issues[slow_performance_fingerprints[0]]
+        sdk_span.set_tag("_performance_issue_slow_span", first_slow_span["span_id"])
+
+    sequential_span_performance_issues = detectors[DetectorType.SEQUENTIAL_SLOW_SPANS].stored_issues
+    sequential_performance_fingerprints = list(sequential_span_performance_issues.keys())
+    if sequential_performance_fingerprints:
+        first_sequential_span = sequential_span_performance_issues[
+            sequential_performance_fingerprints[0]
+        ]
+        sdk_span.set_tag("_performance_issue_sequential_span", first_sequential_span["span_id"])
+
+
+# Creates a stable fingerprint given the same span details using sha1.
+def fingerprint_span(span: Span):
+    op = span.get("op", None)
+    description = span.get("description", None)
+    if not description or not op:
+        return None
+
+    signature = (str(op) + str(description)).encode("utf-8")
+    full_fingerprint = hashlib.sha1(signature).hexdigest()
+    fingerprint = full_fingerprint[
+        :20
+    ]  # 80 bits. Not a cryptographic usage, we don't need all of the sha1 for collision detection
+
+    return fingerprint
+
+
+# Simple fingerprint for broader checks, using the span op.
+def fingerprint_span_op(span: Span):
+    op = span.get("op", None)
+    if not op:
+        return None
+    return op
+
+
+def get_span_duration(span: Span):
+    return span.get("timestamp", 0) - span.get("start_timestamp", 0)
+
+
+class PerformanceDetector(ABC):
+    """
+    Classes of this type have their visit functions called as the event is walked once and will store a performance issue if one is detected.
+    """
+
+    def __init__(self, settings: Dict[str, Any]):
+        self.settings = settings[self.settingsKey]
+        self.init()
+
+    @abstractmethod
+    def init(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def settingsKey(self) -> DetectorType:
```

Comment:
why camelCase?

Reply:
Switching between js and python too much, my bad 😄